src/axolotl/prompt_strategies/completion.py

```diff
@@ -82,11 +96,23 @@ def load(tokenizer, cfg, ds_cfg: Optional[Dict[str, Any]] = None):
     strat = CompletionPromptTokenizingStrategy(
         CompletionPrompter(),
         tokenizer,
-        cfg.train_on_inputs,
+        True,
         cfg.sequence_len,
         max_length=cfg.sequence_len * 64,
     )
     if ds_cfg and "field" in ds_cfg:
         strat.field = ds_cfg["field"]
 
     return strat
+
+
+def load_json(tokenizer, cfg):
+    strat = CompletionJSONPromptTokenizationStrategy(
+        CompletionPrompter(),
+        tokenizer,
+        True,
+        cfg.sequence_len,
+        max_length=cfg.sequence_len * 64,
+    )
+
```

Comment:
Seems like you're missing

```
    if ds_cfg and "field" in ds_cfg:
         strat.field = ds_cfg["field"]
```

 here as well. 

It would be good to not have to repeat this code though. Maybe it can be inherited from a base?